### PR TITLE
FIX: ValueError when hexbin is run with empty arrays and log scaling.

### PR DESCRIPTION
**Pull request by erykoff**
## PR Checklist

**Tests and Styling**
- [ ] Has pytest style unit tests (and `pytest` passes).
- [ ] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).

**Documentation**
- [ ] New features are documented, with examples if plot related.
- [ ] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [ ] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).

### CODE REVIEW

**Comment by erykoff** [comment]
This PR addresses #23922

File: lib/matplotlib/colors.py [hunk 1]
```diff
@@ -1368,8 +1368,14 @@ def inverse(self, value):
     def autoscale(self, A):
         """Set *vmin*, *vmax* to min, max of *A*."""
         A = np.asanyarray(A)
-        self.vmin = A.min()
-        self.vmax = A.max()
+        # Ensure that A has non-zero size to compute min/max.
+        # Otherwise put in filler values.
+        if A.size:
```
Comment (tacaswell) [inline lib/matplotlib/colors.py:1373]: This is hitting _alot_ of use cases as all of the color mappables to through here, would it be possible to do this check in `hexbin` instead?

If we do this here, do we want to use the same logic from the auto limit on the Axis to sure min != max (which I forsee causing divide by 0 errors later)?

File: lib/matplotlib/colors.py [hunk 2]
```diff
@@ -1368,8 +1368,14 @@ def inverse(self, value):
...
+            self.vmin = A.min()
+            self.vmax = A.max()
+        else:
+            self.vmin = 0.0
+            self.vmax = 0.0
```
Comment (anntzer) [inline lib/matplotlib/colors.py:1378]: Setting these values to zero seems almost certainly wrong; the pre-#21989 behavior was to set them to np.ma.masked (I think?) which also seems a bit strange but already better.

**Comment by erykoff** [comment]
Please take a look now.  I pushed the fix further upstream and also used `masked` instead of `0.0`.  Maybe you have some more elegant ideas; however, this does fix the problem and pass the additional test without touching non-hexbin optimized code paths.

**Review by timhoffm — APPROVED** [review]

**Comment by timhoffm** [comment]
Thanks @erykoff and congratulations on your first contribution to Matplotlib.

**Pull request merged**